src/main/java/com/semmle/jira/addon/util/JiraUtils.java

```diff
@@ -138,4 +148,90 @@ public static void createLgtmIssueType() {
+        ComponentAccessor.getComponentOfType(ManagedConfigurationItemService.class);
+
+    CustomField customField = null;
+    ManagedConfigurationItem managedField;
+
+    for (CustomField candidateField :
+        customFieldManager.getCustomFieldObjectsByName(Constants.CUSTOM_FIELD_NAME)) {
+
+      managedField = managedConfigurationItemService.getManagedCustomField(candidateField);
+
+      if (managedField != null
+          && managedField.isManaged()
+          && managedField.getConfigurationItemAccessLevel() == ConfigurationItemAccessLevel.LOCKED
+          && managedField.getSourceId() == Constants.PLUGIN_KEY) {
+        customField = candidateField;
+        break;
+      }
+    }
+
+    if (customField == null) {
+
+      customField =
+          customFieldManager.createCustomField(
+              Constants.CUSTOM_FIELD_NAME,
+              Constants.CUSTOM_FIELD_NAME,
+              customFieldManager.getCustomFieldType(
+                  "com.atlassian.jira.plugin.system.customfieldtypes:textfield"),
+              customFieldManager.getCustomFieldSearcher(
+                  "com.atlassian.jira.plugin.system.customfieldtypes:exacttextsearcher"),
+              Collections.singletonList(GlobalIssueContext.getInstance()),
+              Collections.singletonList(JiraUtils.getLgtmIssueType()));
+
+      managedField = managedConfigurationItemService.getManagedCustomField(customField);
+
+      ManagedConfigurationItemBuilder builder =
+          ManagedConfigurationItemBuilder.builder(managedField);
+
+      builder.setManaged(true);
+      builder.setConfigurationItemAccessLevel(ConfigurationItemAccessLevel.LOCKED);
+      builder.setSource(Constants.PLUGIN_KEY);
+
+      managedConfigurationItemService.updateManagedConfigurationItem(builder.build());
+    }
+
+    getSettingsObject()
+        .put(Constants.CUSTOM_FIELD_CONFIG_KEY, customField.getIdAsLong().toString());
+
+    return;
+  }
+
+  private static PluginSettings getSettingsObject() {
+
+    PluginSettingsFactory pluginSettingsFactory =
+        ComponentAccessor.getOSGiComponentInstanceOfType(PluginSettingsFactory.class);
+
+    PluginSettings settings = pluginSettingsFactory.createGlobalSettings();
+
+    return settings;
+  }
+
+  public static CustomField getConfigKeyCustomField() throws CustomFieldRetrievalException {
+
+    long fieldId;
+    try {
+      fieldId = Long.parseLong((String) getSettingsObject().get(Constants.CUSTOM_FIELD_CONFIG_KEY));
+    } catch (ClassCastException | NumberFormatException e) {
+      log.error("Invalid custom field config key.", e);
+      throw new CustomFieldRetrievalException();
+    }
+
+    CustomField customField =
+        ComponentAccessor.getCustomFieldManager().getCustomFieldObject(fieldId);
+
+    if (customField == null) {
+      log.error("Custom field not found with specified id = " + String.valueOf(fieldId));
+      throw new CustomFieldRetrievalException();
```

Comment:
please include the message in the exception, and don't log the error here yet. Better log the exception where `getConfigKeyCustomField` is called.